src-graphql/product-query.js

```diff
@@ -0,0 +1,39 @@
+import optionQuery from './option-query';
+import imagesQuery from './images-query';
+import variantQuery from './variant-query';
+import createGid from './create-gid';
+
+export default function productQuery({client, id}, fields = ['id', 'createdAt', 'updatedAt', 'bodyHtml', 'handle', 'productType',
+  'title', 'vendor', 'tags', 'publishedAt', ['options', optionQuery()], ['images', imagesQuery()], ['variants', variantQuery()]]) {
+  let query;
+
+  if (id) {
```

Comment:
This might be better split up into two separate query functions. One would takes `id` and one which doesn't.

@minasmart thoughts?